File: ios/Classes/Serializer/ConsentSerializer.swift
import Usercentrics

extension UsercentricsServiceConsent {
    func serialize() -> Any {
        return [
            "templateId": self.templateId,
            "status": self.status,
            "type": self.type?.name as Any,
            "version": self.version,
            "dataProcessor": self.dataProcessor,
            "isEssential": self.isEssential,
            "history": self.history.map { $0.serialize() },
            "category": self.category
        ]
    }
}

extension UsercentricsConsentHistoryEntry {
    func serialize() -> Any {
        return [
            "status": self.status,
            "timestampInMillis": self.timestampInMillis,
            "type": self.type.name as Any,
        ]
    }
}

File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/ConsentSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.sdk.UsercentricsConsentHistoryEntry
import com.usercentrics.sdk.UsercentricsServiceConsent

internal fun UsercentricsServiceConsent.serialize(): Any {
    return mapOf(
        "templateId" to templateId,
        "status" to status,
        "type" to type?.name,
        "version" to version,
        "dataProcessor" to dataProcessor,
        "isEssential" to isEssential,
        "history" to history.map { it.serialize() },
        "category" to category
    )
}

private fun UsercentricsConsentHistoryEntry.serialize(): Any {
    return mapOf(
        "status" to status,
        "timestampInMillis" to timestampInMillis,
        "type" to type.name
    )
}